jmx-metrics/src/main/resources/target-systems/wildfly.groovy

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+def beanWildflyDeployment = otel.mbeans("jboss.as.expr:deployment=*,subsystem=undertow")
+otel.instrument(beanWildflyDeployment, "wildfly.session.count", "The number of sessions created.", "{sessions}",
+  ["deployment": { mbean -> mbean.name().getKeyProperty("deployment")}],
+  "sessionsCreated", otel.&longCounterCallback)
```

Comment:
Would it make sense to use the new attribute map feature and have all of these with the `wildfly.session.count` instrument name and a ~`state` attribute?

```groovy
 otel.instrument(
  beanWildflyDeployment, "wildfly.session.count",
  "The number of sessions in a given state", "{sessions}",
  ["deployment" : { mbean -> mbean.name().getKeyProperty("deployment") }],
  ["sessionsCreated": ["state": {mbean -> "created"}],
   "activeSessions": ["state": {mbean -> "active"}],
   "expiredSessions": ["state": {mbean -> "expired"}],
   "rejectedSessions": ["state": {mbean -> "rejected"}]],
  otel.&longCounterCallback)
```

Reply:
I originally wanted to data model these after that pattern, but I think the main issue I encountered is that "activeSessions" is a non-monotonic sum. It represents what is currently active. In contrast, created, expired, and rejected are monotonic. If we were to create a single metric and attribute it by those states, the sum of everything would not be meaningful.

Reply:
Oh right, though I wonder if the others would still be applicable for this, but the names are pretty clear as is.